pkg/osvscanner/osvscanner.go

```diff
@@ -399,6 +406,33 @@ func scanLockfile(r reporter.Reporter, path string, parseAs string) ([]scannedPa
 	return packages, nil
 }
 
+func extractMavenDeps(f lockfile.DepFile) (lockfile.Lockfile, error) {
+	depClient, err := client.NewDepsDevClient(depsdev.DepsdevAPI)
+	if err != nil {
+		return lockfile.Lockfile{}, err
+	}
+	extractor := manifest.MavenResolverExtractor{
+		DependencyClient: depClient,
+	}
+	packages, err := extractor.Extract(f)
+	if err != nil {
+		err = fmt.Errorf("failed extracting %s: %w", f.Path(), err)
+	}
+	sort.Slice(packages, func(i, j int) bool {
```

Comment:
I assume this sorting is just for test consistency? If so, let's add a comment saying so.